ricecooker/classes/nodes.py

```diff
@@ -171,31 +181,39 @@ class ContentNode(Node):
             files (str or list): content's associated file(s)
             thumbnail (str): local path or url to thumbnail image (optional)
     """
-    def __init__(self, *args, **kwargs):
+    def __init__(self, id, title, description="", author="", files=None, thumbnail=None, license=None, questions=None, extra_fields=None):
         # Map parameters to model variables
-        assert isinstance(args[0], str), "id must be a string"
-        self.id = args[0]
-        self.original_id = args[0]
-        self.title = args[1]
-        self.description = kwargs.get('description') or ""
-        self.author = kwargs.get('author') or ""
-        self.license = kwargs.get('license')
+        assert isinstance(id, str), "id must be a string"
+        self.source_id = id
+        self.title = title
+        self.description = description or ""
+        self.author = author or ""
+        self.license = license
 
         # Set files into list format (adding thumbnail if provided)
-        files = kwargs.get('files') or []
-        self.files = [files] if isinstance(files, str) else files
-        self.thumbnail = kwargs.get('thumbnail')
+        self.files = files or []
+        self.thumbnail = thumbnail
 
         # Set any possible exercise data to standard format
-        self.questions = kwargs.get('questions') or []
-        self.extra_fields = kwargs.get('extra_fields') or {}
+        self.questions = questions or []
+        self.extra_fields = extra_fields or {}
         super(ContentNode, self).__init__()
 
     def __str__(self):
         count = self.count()
         metadata = "{0} {1}".format(count, "descendant" if count == 1 else "descendants")
         return "{title} ({kind}): {metadata}".format(title=self.title, kind=self.__class__.__name__, metadata=metadata)
 
+    def get_domain_namespace(self):
+        return self.parent.get_domain_namespace()
```

Comment:
(Shared params like that can be handled in the base class and just passed through as part of `**kwargs`